lasagne/layers/noise.py

```diff
@@ -15,12 +17,58 @@
 
 
 class DropoutLayer(Layer):
-    def __init__(self, incoming, p=0.5, rescale=True, **kwargs):
-        super(DropoutLayer, self).__init__(incoming, **kwargs)
+    """Dropout layer [1]_,[2]_
+
+    Sets values to zero with probability p. See notes for disabling dropout
+    during testing.
+
+    Parameters
+    ----------
+    input_layer : `Layer` instance
+        The layer from which this layer will obtain its input
+    p : float or scalar tensor
+        The probability of setting a value to zero
+
+    Notes
+    ----------
+    The dropout layer is a regularizer that randomly sets input values to
+    zero, see [1,2] for a why this might improve generalization.
+    During training you should set deterministic to false and during
+    testing you should set deterministic to true.
```

Comment:
If I understand the [numpydoc guide](https://github.com/numpy/numpy/blob/master/doc/HOWTO_DOCUMENT.rst.txt) correctly, I think the correct markup would be:
```
During training you should set `deterministic` to ``False`` and during testing you should set `deterministic` to ``False``.
```
I.e., single backticks for referencing argument names, and double backticks for other things to be set in a monospace font.